### ISSUE TITLE
Trailing Whitespace causes format (-F) to fail the first time

### ISSUE THREAD

**Issue by rnett** [pinterest/ktlint#1116]
I have a file [here](https://gist.github.com/rnett/e21f4775e18a8e63d9d37c28c031f73a) that when I run `ktlint -F` on it, it fails with messages like:
```
ktlint:
   [ktlint] /windows/Users/jimne/Desktop/OtherStuff/tensorflow_java/tensorflow-core-kotlin/tensorflow-core-kotlin-api/src/gen/annotations/org/tensorflow/op/kotlin/IoOps.kt:1588:33: Trailing space(s) (no-trailing-spaces)
```

If you run `ktlint` without the `-F` afterwards, you'll see the reported `no-trailing-spaces` again.

However, if you run with `-F` a second time, they will be fixed without issue.

(There were [warning messages](https://gist.github.com/rnett/3524b236816421eef728897d2d437cdd) from JDK encapsulation, but they were derailing the issue)

**Comment by Tapchicoma** [comment]
Seems it is related to this Kotlin issue: https://youtrack.jetbrains.com/issue/KT-44624

**Comment by rnett** [comment]
That's not actually causing the crash though, it's just warnings.

**Comment by Moisan** [comment]
This is raising an error with Java 16.

**Comment by timoknapp** [comment]
@Moisan you are right, this warning actually transformed to an error with Java 16:

How come that you assume its related to homebrew?

```
ktlint -F
Exception in thread "main" java.util.concurrent.ExecutionException: java.lang.ExceptionInInitializerError
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
	at com.pinterest.ktlint.KtlintCommandLine.parallel(Main.kt:575)
	at com.pinterest.ktlint.KtlintCommandLine.parallel$default(Main.kt:547)
	at com.pinterest.ktlint.KtlintCommandLine.lintFiles(Main.kt:298)
	at com.pinterest.ktlint.KtlintCommandLine.run(Main.kt:262)
	at com.pinterest.ktlint.Main.main(Main.kt:70)
Caused by: java.lang.ExceptionInInitializerError
	at org.jetbrains.kotlin.com.intellij.pom.java.LanguageLevel.<clinit>(LanguageLevel.java:25)
	at org.jetbrains.kotlin.com.intellij.core.CoreLanguageLevelProjectExtension.<init>(CoreLanguageLevelProjectExtension.java:26)
	at org.jetbrains.kotlin.com.intellij.core.JavaCoreProjectEnvironment.<init>(JavaCoreProjectEnvironment.java:42)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreProjectEnvironment.<init>(KotlinCoreProjectEnvironment.kt:26)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment$ProjectEnvironment.<init>(KotlinCoreEnvironment.kt:121)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment$Companion.createForProduction(KotlinCoreEnvironment.kt:425)
	at com.pinterest.ktlint.core.internal.KotlinPsiFileFactory.initializePsiFileFactory(KotlinPsiFileFactory.kt:74)
	at com.pinterest.ktlint.core.internal.KotlinPsiFileFactory.acquirePsiFileFactory$ktlint_core(KotlinPsiFileFactory.kt:39)
	at com.pinterest.ktlint.core.KtLint.format(KtLint.kt:378)
	at com.pinterest.ktlint.internal.FileUtilsKt.formatFile(FileUtils.kt:182)
	at com.pinterest.ktlint.KtlintCommandLine.process(Main.kt:366)
	at com.pinterest.ktlint.KtlintCommandLine.access$process(Main.kt:89)
	at com.pinterest.ktlint.KtlintCommandLine$lintFiles$3$1.call(Main.kt:289)
	at com.pinterest.ktlint.KtlintCommandLine$lintFiles$3$1.call(Main.kt:288)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:831)
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make protected void java.util.ResourceBundle.setParent(java.util.ResourceBundle) accessible: module java.base does not "opens java.util" to unnamed module @76e041cf
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:357)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
	at java.base/java.lang.reflect.Method.checkCanSetAccessible(Method.java:199)
	at java.base/java.lang.reflect.Method.setAccessible(Method.java:193)
	at org.jetbrains.kotlin.com.intellij.util.ReflectionUtil.makeAccessible(ReflectionUtil.java:252)
	at org.jetbrains.kotlin.com.intellij.util.ReflectionUtil.getDeclaredMethod(ReflectionUtil.java:269)
	at org.jetbrains.kotlin.com.intellij.DynamicBundle.<clinit>(DynamicBundle.java:22)
	... 18 more
```

**Comment by rnett** [comment]
This is a known issue caused by JDK 16 encapsulating it's internals and modules more strongly, it should be fixed with 1.5 (well, once ktlint uses 1.5).  It's entirely separate from the issue I originally raised though (at least, I'm 99% sure it is).

**Comment by paul-dingemans** [comment]
The problem is caused in the snippet below:
```
    @JvmName("decodeRawReified")
    public inline fun <reified T : TType> decodeRaw(bytes: Operand<TString>, littleEndian: Boolean?
            = null): DecodeRaw<T> = decodeRaw<T>(bytes, T::class.java, littleEndian)
```

The first run of `ktlint --format --verbose --relative --experimental` results in output:
```
src/main/kotlin/Issue1116.kt:3:11: Trailing space(s) (no-trailing-spaces)
```
and the result files (of course without the added comment) becomes:
```
@JvmName("decodeRawReified")
public inline fun <reified T : TType> decodeRaw(
    bytes: Operand<TString>,
    littleEndian: Boolean? =    // <-- Note that here are still some trailing spaces 
        null
): DecodeRaw<T> = decodeRaw<T>(bytes, T::class.java, littleEndian)
```

The second run of `ktlint --format --verbose --relative --experimental` results in output:
```
@JvmName("decodeRawReified")
public inline fun <reified T : TType> decodeRaw(
    bytes: Operand<TString>,
    littleEndian: Boolean? = // <-- Note that the trailing spaces are now gone
        null
): DecodeRaw<T> = decodeRaw<T>(bytes, T::class.java, littleEndian)
```

When running `ktlint --verbose --relative --experimental` on the first example of this comment, the output is as follows:
```
src/main/kotlin/Issue1116.kt:2:49: Parameter should be on a separate line (unless all parameters can fit a single line) (parameter-list-wrapping)
src/main/kotlin/Issue1116.kt:2:49: Missing newline after "(" (indent)
src/main/kotlin/Issue1116.kt:2:73: Missing newline after "," (indent)
src/main/kotlin/Issue1116.kt:2:74: Parameter should be on a separate line (unless all parameters can fit a single line) (parameter-list-wrapping)
src/main/kotlin/Issue1116.kt:3:1: Unexpected indentation (6) (should be 4) (indent)
src/main/kotlin/Issue1116.kt:3:7: Line break before assignment is not allowed (no-line-break-before-assignment)
src/main/kotlin/Issue1116.kt:3:12: Missing newline before ")" (indent)
src/main/kotlin/Issue1116.kt:3:13: Missing newline before ")" (parameter-list-wrapping)
```
it is noteworthy that the trailing spaces are not reported above. But we can see that at line 3, several other rules report a problem. I expect that the `parameter-list-wrapping` rule is responsible for formatting the second code sample in this comment. 

Running `ktlint --format --verbose --relative --experimental --disabled_rules=parameter-list-wrapping` on the first example of this comment again, the output is as follows:
```
@JvmName("decodeRawReified")
public inline fun <reified T : TType> decodeRaw(
    bytes: Operand<TString>,
    littleEndian: Boolean? = // No trailing spaces found at the *first run*
        null
): DecodeRaw<T> = decodeRaw<T>(bytes, T::class.java, littleEndian)
```

So the problem should be solved in the `parameter-list-wrapping`. When inserting a newline, it should be more careful not to insert trailing spaces as well.

**Comment by paul-dingemans** [comment]
This problem i(at least with the snippet from my analysis of December 2021) is not reproducable on versions 0.45.x and 0.46.x.
